primitives/runtime-interface/src/impls.rs

```diff
@@ -318,9 +318,8 @@ macro_rules! impl_traits_for_arrays {
 				type SelfInstance = [u8; $n];
 
 				fn from_ffi_value(context: &mut dyn FunctionContext, arg: u32) -> Result<[u8; $n]> {
-					let data = context.read_memory(Pointer::new(arg), $n)?;
 					let mut res = [0u8; $n];
-					res.copy_from_slice(&data);
+					context.read_memory_into(Pointer::new(arg), &mut res)?;
```

Comment:
could use MaybeUninit here? though maybe the compiler is clever enough to omit the zeroing - you'd need to check godbolt to be sure.

Reply:
Yeah, it might be possible to omit the zeroing if we'd use `MaybeUninit`, although that would require modifying the trait, as otherwise it's technically UB. (UB with which I very much disagree, but my personal opinion doesn't really matter here.)

I just wanted to push this quick and easy fix without doing anything potentially controversial.